### PR TITLE
google.cloud.pubsub changed a bunch in 0.29  …

### PR DESCRIPTION
**Pull request by zoidyzoidzoid**
https://github.com/GoogleCloudPlatform/google-cloud-python/releases/tag/pubsub-0.29.0

### CODE REVIEW

File: pubsub/protocol.py [hunk 1]
```diff
@@ -17,7 +22,44 @@ def publish(self, topic, message):
         serialized = self.serializer.encode(message=message)
         self.adapter.publish(topic, serialized)
 
-    def subscribe(self, topic):
-        for message in self.adapter.subscribe(topic):
-            serialized = self.serializer.decode(message)
-            yield serialized
+    def subscribe(self, topic, callback=None, exception_handler=lambda x, y: None, always_raise=True):
```
Comment (BradWhittington) [inline pubsub/protocol.py]: Shouldn't this signature be extended to all pubsub drivers?

Comment (zoidyzoidzoid) [reply pubsub/protocol.py]: Definitely

**Comment by BradWhittington** [comment]
Please update README examples... and include stuff about how you'd include SENTRY in the example in README...

File: pubsub/adapters/googlecloud.py [hunk 2]
```diff
@@ -1,52 +1,87 @@
...
-            topic.create()
-        topic.publish(message)
+        topic_path = self.publisher.topic_path(self.project_id, topic_name)
+        exists = True
+        try:
+            self.publisher.get_topic(topic_path)
+        except GaxError as exc:
+            if exc.cause._state.code != StatusCode.NOT_FOUND:
+                raise
+            exists = False
```
Comment (bertuss) [inline pubsub/adapters/googlecloud.py]: Can we not do `self.publisher.create_topic(topic_path)` here without the need for the boolean exists?

**Review by bertuss — APPROVED** [review]
For what it's worth, it looks good to me.

File: pubsub/adapters/googlecloud.py (hunk 2 above)
Comment (BradWhittington) [reply pubsub/adapters/googlecloud.py]: 👍

**Pull request merged**